src/vs/workbench/contrib/chat/browser/chatEditing/chatEditingTextModelChangeService.ts

```diff
@@ -0,0 +1,400 @@
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { addDisposableListener, getWindow } from '../../../../../base/browser/dom.js';
+import { assert } from '../../../../../base/common/assert.js';
+import { DeferredPromise, RunOnceScheduler, timeout } from '../../../../../base/common/async.js';
+import { Emitter } from '../../../../../base/common/event.js';
+import { Disposable, toDisposable } from '../../../../../base/common/lifecycle.js';
+import { IObservable, observableValue } from '../../../../../base/common/observable.js';
+import { isEqual } from '../../../../../base/common/resources.js';
+import { themeColorFromId } from '../../../../../base/common/themables.js';
+import { assertType } from '../../../../../base/common/types.js';
+import { URI } from '../../../../../base/common/uri.js';
+import { EditOperation, ISingleEditOperation } from '../../../../../editor/common/core/editOperation.js';
+import { StringEdit } from '../../../../../editor/common/core/edits/stringEdit.js';
+import { Range } from '../../../../../editor/common/core/range.js';
+import { IDocumentDiff, nullDocumentDiff } from '../../../../../editor/common/diff/documentDiffProvider.js';
+import { DetailedLineRangeMapping } from '../../../../../editor/common/diff/rangeMapping.js';
+import { TextEdit } from '../../../../../editor/common/languages.js';
+import { IModelDeltaDecoration, ITextModel, ITextSnapshot, MinimapPosition, OverviewRulerLane } from '../../../../../editor/common/model.js';
+import { ModelDecorationOptions } from '../../../../../editor/common/model/textModel.js';
+import { offsetEditFromContentChanges, offsetEditFromLineRangeMapping, offsetEditToEditOperations } from '../../../../../editor/common/model/textModelStringEdit.js';
+import { IEditorWorkerService } from '../../../../../editor/common/services/editorWorker.js';
+import { TextModelEditReason } from '../../../../../editor/common/textModelEditReason.js';
+import { IModelContentChangedEvent } from '../../../../../editor/common/textModelEvents.js';
+import { AccessibilitySignal, IAccessibilitySignalService } from '../../../../../platform/accessibilitySignal/browser/accessibilitySignalService.js';
+import { editorSelectionBackground } from '../../../../../platform/theme/common/colorRegistry.js';
+import { ICellEditOperation } from '../../../notebook/common/notebookCommon.js';
+import { ModifiedFileEntryState } from '../../common/chatEditingService.js';
+import { IChatResponseModel } from '../../common/chatModel.js';
+import { IDocumentDiff2 } from './chatEditingCodeEditorIntegration.js';
+import { pendingRewriteMinimap } from './chatEditingModifiedFileEntry.js';
+
+
+export class ChatEditingTextModelChangeService extends Disposable {
```

Comment:
Moved all of the code related to 
* Updating original model (restore snapshot)
* Updating modified model (restore snapshot, accept/reject, mirror edits, accept/reject chunks etc)
* Decorations
* Computing diff

Basically tried to move all of the text model(s) related stuff into this class.